app/src/main/java/com/codingblocks/cbonlineapp/util/DeleteDownloadedFiles.kt

```diff
@@ -0,0 +1,36 @@
+package com.codingblocks.cbonlineapp.util
+
+import android.content.Context
+import android.os.Environment
+import androidx.work.CoroutineWorker
+import androidx.work.WorkerParameters
+import com.codingblocks.cbonlineapp.database.ContentDao
+import com.codingblocks.cbonlineapp.database.models.ContentModel
+import org.koin.core.KoinComponent
+import org.koin.core.inject
+import java.io.File
+import java.util.*
+
+class DeleteDownloadedFiles(context: Context, private val workerParameters: WorkerParameters) :
+    CoroutineWorker(context, workerParameters), KoinComponent {
+
+    val contentDao: ContentDao by inject()
+
+    override suspend fun doWork(): Result {
+        val list:List<ContentModel> = contentDao.getDownloads(true)
+
+        list.forEach {item->
+            val folderFile = File(applicationContext.getExternalFilesDir(Environment.getDataDirectory().absolutePath), "/${item.contentLecture.lectureId}")
+            val fileDate = Date(folderFile.lastModified())
+            val expiringDate = Date(fileDate.time + 15)
```

Comment:
@prathamk22 
```
            val calendar = Calendar.getInstance()
            calendar.time = Date(fileDate.time)
            calendar.add(Calendar.DATE, 14)
            val expiringDate = calendar.time
```

it was supposed to be done like this

Reply:
Okay sir.